src/gathernodeinputhandler.cpp

```diff
@@ -62,36 +63,41 @@ Status GatherNodeInputHandler::notifyFinishedDependency() {
         const auto shardsCount = shardMap.size();
         SPDLOG_LOGGER_DEBUG(dag_executor_logger, "Consolidating: {} shards for input: {}", shardsCount, inputName);
         session_id_t firstShardId = 0;
-        auto firstShardTensorDesc = shardMap.at(firstShardId)->getTensorDesc();
-        auto shardDims = getEffectiveShape(firstShardTensorDesc);
-        auto newDims = shardDims;
+        auto firstShard = shardMap.at(firstShardId);
+        auto firstShardDims = firstShard->get_shape();
+        auto precision = firstShard->get_element_type();
+        auto newDims = firstShardDims;
         newDims.insert(newDims.begin(),
             collapsingDetails->collapsedSessionSizes.begin(),
             collapsingDetails->collapsedSessionSizes.end());
-        const InferenceEngine::TensorDesc consolidatedBlobDesc(
-            firstShardTensorDesc.getPrecision(),
-            newDims,
-            InferenceEngine::Layout::ANY);
-        InferenceEngine::Blob::Ptr consolidatedBlob;
-        auto status = createSharedBlob(consolidatedBlob, consolidatedBlobDesc);
+        std::shared_ptr<ov::runtime::Tensor> consolidatedTensor;
+        auto status = createSharedTensor(consolidatedTensor, precision, newDims);
         if (!status.ok()) {
             return status;
         }
-        for (auto& [shardId, blob] : shardMap) {
-            auto& shardTensorDesc = blob->getTensorDesc();
-            if (shardTensorDesc != firstShardTensorDesc) {
-                SPDLOG_LOGGER_ERROR(dag_executor_logger, "Failed to consolidate blob: {} shards in gather node. First shard has different tensor description: {} than current shard: {}",
+        for (auto& [shardId, tensor] : shardMap) {
+            if ((tensor->get_element_type() != precision) ||
+                (tensor->get_shape() != firstShardDims)) {
+                std::stringstream firstShardShapeStream;
+                firstShardShapeStream << firstShardDims;
+                auto currentShardShape = tensor->get_shape();
+                std::stringstream currentShardShapeStream;
+                currentShardShapeStream << currentShardShape;
+                SPDLOG_LOGGER_ERROR(dag_executor_logger, "Failed to consolidate tensor: {}; shards in gather node. First shard has different tensor precision:{};  or shape: {}; than current shard precision: {}; shape:{};",
```

Comment:
Inconsistency with spaces, should be probbably: Failed to consolidate tensor: {}; shards in gather node. First shard has different tensor precision: {}; or shape: {}; than current shard precision: {}; shape: {};